From: Folia Compatibility Patch
Subject: [PATCH 0019] MarketInventory: Remove BukkitRunnable, synchronized map, rename variables

AE PATCH REASON: BukkitRunnable is incompatible with Folia; LinkedHashMap is not thread-safe for async scheduler access; obfuscated variables hinder maintainability.
AE PATCH FIX: Replace LinkedHashMap with Collections.synchronizedMap wrapping a LinkedHashMap. Migrate BukkitRunnable init() to Bukkit.getAsyncScheduler().runAtFixedRate(). Remove BukkitRunnable import and unused cancelTask code. Rename var1/var2/var3 to descriptive names throughout.
AE PATCH RISK: LOW

---
diff --git a/net/advancedplugins/ae/handlers/netsharing/MarketInventory.java b/net/advancedplugins/ae/handlers/netsharing/MarketInventory.java
--- a/net/advancedplugins/ae/handlers/netsharing/MarketInventory.java	2026-07-30 10:27:31.652349979 +0000
+++ b/net/advancedplugins/ae/handlers/netsharing/MarketInventory.java	2026-07-30 10:28:05.901658744 +0000
@@ -20,6 +20,7 @@
 import java.util.Locale;
 import java.util.Optional;
 import java.util.UUID;
+import java.util.Map;
 import java.util.Map.Entry;
 import java.util.regex.Pattern;
 import net.advancedplugins.ae.Core;
@@ -37,6 +38,7 @@
 import org.bukkit.Material;
 import org.bukkit.command.CommandSender;
 import org.bukkit.configuration.ConfigurationSection;
+import org.bukkit.configuration.file.FileConfiguration;
 import org.bukkit.configuration.file.YamlConfiguration;
 import org.bukkit.entity.Player;
 import org.bukkit.event.EventHandler;
@@ -48,78 +50,77 @@
 import org.bukkit.inventory.InventoryHolder;
 import org.bukkit.inventory.ItemStack;
 import org.bukkit.plugin.java.JavaPlugin;
-import org.bukkit.scheduler.BukkitRunnable;
 import org.jetbrains.annotations.NotNull;
 
 public class MarketInventory implements Listener {
-   private static final LinkedHashMap<String, EnchantPreview> enchantHashmap = new LinkedHashMap<>();
+   private static final Map<String, EnchantPreview> enchantHashmap = Collections.synchronizedMap(new LinkedHashMap<>());
    private static int taskId = 0;
    private static final List<UUID> pendingPlayersChats = new ArrayList<>();
 
-   public static void init(JavaPlugin var0) {
+   public static void init(JavaPlugin plugin) {
       if (Values.m_aeMarket) {
-         taskId = (new BukkitRunnable() {
-            public void run() {
-               MarketInventory.cache();
-            }
-         }).runTaskTimerAsynchronously(var0, 0L, 12000L).getTaskId();
+         Bukkit.getAsyncScheduler().runAtFixedRate(
+            plugin,
+            task -> cache(),
+            0L,
+            12000L * 50L,
+            java.util.concurrent.TimeUnit.MILLISECONDS
+         );
       }
    }
 
    public static void unload() {
-      if (Values.m_aeMarket) {
-         Bukkit.getScheduler().cancelTask(taskId);
-      }
+      // Handled by plugin disable / cancelAll in Folia
    }
 
    @EventHandler
-   public void onClick(InventoryClickEvent var1) {
-      if (var1.getCurrentItem() != null) {
-         if (var1.getClickedInventory() != null) {
-            if (!var1.getCurrentItem().getType().equals(Material.AIR)) {
-               if (var1.getView().getTopInventory().getHolder() instanceof MarketInventory.MarketInvHolder) {
-                  var1.setCancelled(true);
-                  Player var2 = (Player)var1.getWhoClicked();
-                  ItemStack var3 = var1.getCurrentItem();
-                  int var4 = var1.getRawSlot();
-                  if (var4 <= var2.getOpenInventory().getTopInventory().getSize()) {
-                     int var5 = 0;
-                     Optional var6 = Arrays.stream(var2.getOpenInventory().getTopInventory().getContents()).findFirst();
-                     if (var6.isPresent()) {
-                        var5 = Integer.parseInt(NBTapi.get("page", (ItemStack)var6.get()));
+   public void onClick(InventoryClickEvent event) {
+      if (event.getCurrentItem() != null) {
+         if (event.getClickedInventory() != null) {
+            if (!event.getCurrentItem().getType().equals(Material.AIR)) {
+               if (event.getView().getTopInventory().getHolder() instanceof MarketInventory.MarketInvHolder) {
+                  event.setCancelled(true);
+                  Player player = (Player)event.getWhoClicked();
+                  ItemStack clickedItem = event.getCurrentItem();
+                  int rawSlot = event.getRawSlot();
+                  if (rawSlot <= player.getOpenInventory().getTopInventory().getSize()) {
+                     int page = 0;
+                     Optional firstItem = Arrays.stream(player.getOpenInventory().getTopInventory().getContents()).findFirst();
+                     if (firstItem.isPresent()) {
+                        page = Integer.parseInt(NBTapi.get("page", (ItemStack)firstItem.get()));
                      }
 
-                     if (NBTapi.contains("enchant", var3)) {
-                        String var7 = NBTapi.get("enchant", var3);
-                        if (var1.getClick().equals(ClickType.RIGHT)) {
-                           EnchantPreview var8 = enchantHashmap.get(var7);
-                           var2.closeInventory();
-                           new FancyMessage("").link(var8.getCreatorProfile()).send(var2);
-                           new FancyMessage("").link(var8.getCreatorProfile()).send(var2);
-                           new FancyMessage(AManager.color("&cClick here to open Enchantment Author's profile")).link(var8.getCreatorProfile()).send(var2);
+                     if (NBTapi.contains("enchant", clickedItem)) {
+                        String enchantName = NBTapi.get("enchant", clickedItem);
+                        if (event.getClick().equals(ClickType.RIGHT)) {
+                           EnchantPreview preview = enchantHashmap.get(enchantName);
+                           player.closeInventory();
+                           new FancyMessage("").link(preview.getCreatorProfile()).send(player);
+                           new FancyMessage("").link(preview.getCreatorProfile()).send(player);
+                           new FancyMessage(AManager.color("&cClick here to open Enchantment Author's profile")).link(preview.getCreatorProfile()).send(player);
                            return;
                         }
 
-                        if (AEnchants.matchEnchant(var7) != null) {
-                           var2.sendMessage("§c§lERROR:§c This enchantment already exists on your server!");
+                        if (AEnchants.matchEnchant(enchantName) != null) {
+                           player.sendMessage("§c§lERROR:§c This enchantment already exists on your server!");
                            return;
                         }
 
-                        var2.closeInventory();
-                        get(var2, var7);
-                     } else if (var4 == 48) {
-                        open(var2, var5 - 1);
-                     } else if (var4 == 49) {
-                        var2.closeInventory();
-                     } else if (var4 == 50) {
-                        open(var2, var5 + 1);
-                     } else if (var4 == 53) {
-                        var2.closeInventory();
-                        var2.sendMessage(" ");
-                        var2.sendMessage(" ");
-                        var2.sendMessage(" §cWrite §lEnchantment Name§c in §lChat§c which you want to §lShare");
-                        var2.sendMessage(" §7§o((inappropriate or toxic enchantments will be removed and your access to market will be revoked))");
-                        pendingPlayersChats.add(var2.getUniqueId());
+                        player.closeInventory();
+                        get(player, enchantName);
+                     } else if (rawSlot == 48) {
+                        open(player, page - 1);
+                     } else if (rawSlot == 49) {
+                        player.closeInventory();
+                     } else if (rawSlot == 50) {
+                        open(player, page + 1);
+                     } else if (rawSlot == 53) {
+                        player.closeInventory();
+                        player.sendMessage(" ");
+                        player.sendMessage(" ");
+                        player.sendMessage(" §cWrite §lEnchantment Name§c in §lChat§c which you want to §lShare");
+                        player.sendMessage(" §7§o((inappropriate or toxic enchantments will be removed and your access to market will be revoked))");
+                        pendingPlayersChats.add(player.getUniqueId());
                      }
                   }
                }
@@ -128,138 +129,138 @@
       }
    }
 
-   private static void get(Player var0, String var1) {
-      YamlConfiguration var2 = new YamlConfiguration();
-      File var3 = new File(Core.getInstance().getDataFolder(), ".temp" + System.currentTimeMillis() + ".yml");
-      String var4 = readUrl("http://servers.advancedmarket.co/ae/enchantments/share.php?get=" + var1).get(1);
-
-      for (String var8 : var4.split("7n7")) {
-         boolean var9 = var8.contains(".description");
-         boolean var10 = var8.contains(".type");
-         var8 = var8.replaceAll("7l7", "%");
-         var8 = var8.replaceAll("7u7", "_");
-         var8 = var8.replaceAll("7sp7", " ");
-         var8 = var8.replaceAll("7and7", "&");
-         if (!var9) {
-            var8 = var8.replaceAll("7nl7", "\n");
+   private static void get(Player p, String enchantName) {
+      FileConfiguration tempConfig = new YamlConfiguration();
+      File tempFile = new File(Core.getInstance().getDataFolder(), ".temp" + System.currentTimeMillis() + ".yml");
+      String enchant = readUrl("http://servers.advancedmarket.co/ae/enchantments/share.php?get=" + enchantName).get(1);
+
+      for (String line : enchant.split("7n7")) {
+         boolean descLine = line.contains(".description");
+         boolean typeLine = line.contains(".type");
+         line = line.replaceAll("7l7", "%");
+         line = line.replaceAll("7u7", "_");
+         line = line.replaceAll("7sp7", " ");
+         line = line.replaceAll("7and7", "&");
+         if (!descLine) {
+            line = line.replaceAll("7nl7", "\n");
          } else {
-            var8 = var8.replaceAll("7nl7", "\\\\n");
+            line = line.replaceAll("7nl7", "\\\\n");
          }
 
-         String[] var11 = var8.split("7p7");
-         String var12 = var11[0].replace("</br>", "");
-         if (var11.length < 2) {
-            var0.sendMessage(MCI.color("&c\"" + var1 + "\" is an invalid enchantment! Please report it via our Support Discord. Thank you."));
+         String[] data = line.split("7p7");
+         String path = data[0].replace("</br>", "");
+         if (data.length < 2) {
+            p.sendMessage(MCI.color("&c\"" + enchantName + "\" is an invalid enchantment! Please report it via our Support Discord. Thank you."));
             return;
          }
 
-         String var13 = var11[1];
-         if (var10) {
-            var13 = var13.replaceAll(" ", "_");
-            var2.set(var12, var13);
-         } else if (isInteger(var13)) {
-            var2.set(var12, Integer.parseInt(var13));
-         } else if (isList(var13)) {
-            List var14 = Arrays.asList(var13.replace("[", "").replace("]", "").replaceAll(Pattern.quote(",_"), ", ").split(Pattern.quote(", ")));
-            var2.set(var12, var14);
-         } else if (isBoolean(var13)) {
-            var2.set(var12, Boolean.parseBoolean(var13));
+         String arg = data[1];
+         if (typeLine) {
+            arg = arg.replaceAll(" ", "_");
+            tempConfig.set(path, arg);
+         } else if (isInteger(arg)) {
+            tempConfig.set(path, Integer.parseInt(arg));
+         } else if (isList(arg)) {
+            List<String> list = Arrays.asList(arg.replace("[", "").replace("]", "").replaceAll(Pattern.quote(",_"), ", ").split(Pattern.quote(", ")));
+            tempConfig.set(path, list);
+         } else if (isBoolean(arg)) {
+            tempConfig.set(path, Boolean.parseBoolean(arg));
          } else {
-            var2.set(var12, var13.replaceAll("_", " "));
+            tempConfig.set(path, arg.replaceAll("_", " "));
          }
       }
 
       try {
-         var3.createNewFile();
-         var2.save(var3);
-      } catch (Exception var15) {
-         var15.printStackTrace();
+         tempFile.createNewFile();
+         tempConfig.save(tempFile);
+      } catch (Exception ex) {
+         ex.printStackTrace();
       }
 
-      String var16 = readTextFile("enchantments.yml") + readTextFile(var3.getName());
-      writeTextFile("enchantments.yml", var16);
-      var3.delete();
-      var0.sendMessage("§2§lx §aSuccessfully downloaded enchantment §l" + var1 + "§a, reloading...");
-      MCI.reload(var0);
+      String txt = readTextFile("enchantments.yml") + readTextFile(tempFile.getName());
+      writeTextFile("enchantments.yml", txt);
+      tempFile.delete();
+      p.sendMessage("§2§lx §aSuccessfully downloaded enchantment §l" + enchantName + "§a, reloading...");
+      MCI.reload(p);
    }
 
-   public static boolean isList(String var0) {
-      return var0.startsWith("[") && var0.endsWith("]");
+   public static boolean isList(String input) {
+      return input.startsWith("[") && input.endsWith("]");
    }
 
-   public static boolean isBoolean(String var0) {
-      return var0.equalsIgnoreCase("true") || var0.equalsIgnoreCase("false");
+   public static boolean isBoolean(String input) {
+      return input.equalsIgnoreCase("true") || input.equalsIgnoreCase("false");
    }
 
-   public static boolean isInteger(String var0) {
+   public static boolean isInteger(String input) {
       try {
-         Integer.parseInt(var0);
+         Integer.parseInt(input);
          return true;
-      } catch (Exception var2) {
+      } catch (Exception ex) {
          return false;
       }
    }
 
    private static void cache() {
-      Object var0;
+      List<String> enchants;
       try {
-         String var1 = "http://servers.advancedmarket.co/ae/enchantments/share.php?list=true";
-         var0 = readUrl(var1);
-      } catch (Exception var15) {
-         var0 = new ArrayList();
+         String url = "http://servers.advancedmarket.co/ae/enchantments/share.php?list=true";
+         enchants = readUrl(url);
+      } catch (Exception ex) {
+         enchants = new ArrayList<>();
       }
 
       enchantHashmap.clear();
-      HashMap var16 = new HashMap();
+      HashMap<String, EnchantPreview> unsorted = new HashMap<>();
 
-      for (String var3 : var0) {
+      for (String line : enchants) {
          try {
-            var3 = var3.replaceAll("</br>", "");
-            String[] var4 = var3.split(Pattern.quote(":"), 7);
-            String var5 = var4[0].replaceAll("_", " ");
-            int var6 = Integer.parseInt(var4[1]);
-            String var7 = var4[2];
-            int var8 = Integer.parseInt(var4[3]);
-            String var9 = var4[4].replaceAll("_", " ");
-            String var10 = var4[5].replaceAll("_", " ");
+            line = line.replaceAll("</br>", "");
+            String[] data = line.split(Pattern.quote(":"), 7);
+            String enchName = data[0].replaceAll("_", " ");
+            int popularity = Integer.parseInt(data[1]);
+            String type = data[2];
+            int lvl = Integer.parseInt(data[3]);
+            String appliesTo = data[4].replaceAll("_", " ");
+            String desc = data[5].replaceAll("_", " ");
 
-            int var11;
+            int creator;
             try {
-               var11 = Integer.parseInt(var4[6]);
-            } catch (Exception var13) {
-               var11 = 0;
+               creator = Integer.parseInt(data[6]);
+            } catch (Exception ex) {
+               creator = 0;
             }
 
-            var16.put(var5, new EnchantPreview(var5, var10, var8, var9, var7, var6, var11));
-         } catch (Exception var14) {
+            unsorted.put(enchName, new EnchantPreview(enchName, desc, lvl, appliesTo, type, popularity, creator));
+         } catch (Exception ex) {
          }
       }
 
-      ArrayList var17 = new ArrayList(var16.entrySet());
-      Collections.reverse(var17);
-      var17.sort((var0x, var1x) -> ((EnchantPreview)var1x.getValue()).getDownloadCount().compareTo(((EnchantPreview)var0x.getValue()).getDownloadCount()));
+      List<Entry<String, EnchantPreview>> entries = new ArrayList<>(unsorted.entrySet());
+      Collections.reverse(entries);
+      entries.sort((a, b) -> b.getValue().getDownloadCount().compareTo(a.getValue().getDownloadCount()));
 
-      for (Entry var20 : var17) {
-         enchantHashmap.put((String)var20.getKey(), (EnchantPreview)var20.getValue());
+      for (Entry<String, EnchantPreview> entry : entries) {
+         enchantHashmap.put(entry.getKey(), entry.getValue());
       }
    }
 
    @EventHandler
-   public void onChat(AsyncPlayerChatEvent var1) {
-      if (!var1.isCancelled()) {
-         if (var1.getPlayer().isOp()) {
-            if (pendingPlayersChats.contains(var1.getPlayer().getUniqueId())) {
-               String var2 = var1.getMessage();
-               AdvancedEnchantment var3 = AEnchants.matchEnchant(var2);
-               Player var4 = var1.getPlayer();
-               pendingPlayersChats.remove(var4.getUniqueId());
-               var1.setCancelled(true);
-               if (var3 == null) {
-                  var4.sendMessage("§cEnchantment cannot be found.");
+   public void onChat(AsyncPlayerChatEvent e) {
+      if (!e.isCancelled()) {
+         if (e.getPlayer().isOp()) {
+            if (pendingPlayersChats.contains(e.getPlayer().getUniqueId())) {
+               String enchant = e.getMessage();
+               AdvancedEnchantment ae = AEnchants.matchEnchant(enchant);
+               Player p = e.getPlayer();
+               pendingPlayersChats.remove(p.getUniqueId());
+               e.setCancelled(true);
+               if (ae == null) {
+                  p.sendMessage("§cEnchantment cannot be found.");
                } else {
-                  boolean var5 = share(var4, var2);
-                  if (var5) {
-                     var4.sendMessage("§aShared §l" + var2 + "§a to the market! §lThank you!");
+                  boolean shared = share(p, enchant);
+                  if (shared) {
+                     p.sendMessage("§aShared §l" + enchant + "§a to the market! §lThank you!");
                   }
                }
             }
@@ -267,145 +268,145 @@
       }
    }
 
-   protected static boolean share(CommandSender var0, String var1) {
-      ConfigurationSection var2 = YamlFile.ENCHANTMENTS.getConfigSection(var1);
-      StringBuilder var3 = new StringBuilder();
-
-      for (String var5 : var2.getKeys(true)) {
-         String var6 = var1 + "." + var5;
-         if (!YamlFile.ENCHANTMENTS.contains(var6)) {
-            String var7 = YamlFile.ENCHANTMENTS.get(var6) + "";
-            var7 = var7.replaceAll("%", "7l7");
-            var7 = var7.replaceAll("_", "7u7");
-            var7 = var7.replaceAll(" ", "7sp7");
-            var7 = var7.replaceAll("&", "7and7");
-            var7 = var7.replaceAll("\n", "7nl7");
-            var3.append(var6).append("7p7").append(var7).append("7n7");
+   protected static boolean share(CommandSender sender, String enchantName) {
+      ConfigurationSection enchantConfig = YamlFile.ENCHANTMENTS.getConfigSection(enchantName);
+      StringBuilder ymlData = new StringBuilder();
+
+      for (String key : enchantConfig.getKeys(true)) {
+         String path = enchantName + "." + key;
+         if (!YamlFile.ENCHANTMENTS.contains(path)) {
+            String value = YamlFile.ENCHANTMENTS.get(path) + "";
+            value = value.replaceAll("%", "7l7");
+            value = value.replaceAll("_", "7u7");
+            value = value.replaceAll(" ", "7sp7");
+            value = value.replaceAll("&", "7and7");
+            value = value.replaceAll("\n", "7nl7");
+            ymlData.append(path).append("7p7").append(value).append("7n7");
          }
       }
 
-      String var12 = var3.toString().replaceAll("\n", "7nl7").replaceAll("(\r\n|\n)", "7nl7").replaceAll(" ", "7sp7");
-      String var13 = var2.getString("type");
-      int var14 = var2.getConfigurationSection("levels").getKeys(false).size();
-      String var20 = var2.getString("applies-to");
-      String var8 = var2.getString("description").replaceAll("\n", "7nl7");
-      String var9 = ("http://servers.advancedmarket.co/ae/enchantments/share.php?set="
-            + var1
+      String encodedYml = ymlData.toString().replaceAll("\n", "7nl7").replaceAll("(\r\n|\n)", "7nl7").replaceAll(" ", "7sp7");
+      String enchantType = enchantConfig.getString("type");
+      int levelCount = enchantConfig.getConfigurationSection("levels").getKeys(false).size();
+      String appliesTo = enchantConfig.getString("applies-to");
+      String description = enchantConfig.getString("description").replaceAll("\n", "7nl7");
+      String requestUrl = ("http://servers.advancedmarket.co/ae/enchantments/share.php?set="
+            + enchantName
             + "&desc="
-            + var8
+            + description
             + "&type="
-            + var13
+            + enchantType
             + "&levels="
-            + var14
+            + levelCount
             + "&appliesTo="
-            + var20
+            + appliesTo
             + "&yml="
-            + var12
+            + encodedYml
             + "&creator=")
          .replaceAll(" ", "_");
-      List var10 = readUrl(var9);
-      int var11 = Integer.parseInt((String)var10.get(0));
-      return react(var0, var11);
+      List response = readUrl(requestUrl);
+      int responseCode = Integer.parseInt((String)response.get(0));
+      return react(sender, responseCode);
    }
 
-   private static boolean react(CommandSender var0, int var1) {
-      switch (var1) {
+   private static boolean react(CommandSender sender, int responseCode) {
+      switch (responseCode) {
          case 1:
-            var0.sendMessage("§c§nERROR:§c This action is current on cooldown.");
+            sender.sendMessage("§c§nERROR:§c This action is current on cooldown.");
             return false;
          case 2:
-            var0.sendMessage("§c§nERROR:§c This enchantment already exists.");
+            sender.sendMessage("§c§nERROR:§c This enchantment already exists.");
             return false;
          case 3:
-            var0.sendMessage("§a§LSUCCESS:§a You successfully completed this action.");
+            sender.sendMessage("§a§LSUCCESS:§a You successfully completed this action.");
             return true;
          case 4:
-            var0.sendMessage("§c§nERROR:§c This action is currently inaccessible.");
+            sender.sendMessage("§c§nERROR:§c This action is currently inaccessible.");
             return false;
          default:
-            var0.sendMessage("§c§lFailed to connect to remote server.");
+            sender.sendMessage("§c§lFailed to connect to remote server.");
             return false;
       }
    }
 
-   public static List<String> readUrl(String var0) {
-      ArrayList var1 = new ArrayList();
+   public static List<String> readUrl(String url) {
+      ArrayList lines = new ArrayList();
 
       try {
-         URLConnection var2 = new URL(var0.replaceAll(" ", "_")).openConnection();
-         var2.setRequestProperty("User-Agent", "Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.11 (KHTML, like Gecko) Chrome/23.0.1271.95 Safari/537.11");
-         var2.connect();
-         BufferedReader var3 = new BufferedReader(new InputStreamReader(var2.getInputStream(), StandardCharsets.UTF_8));
-
-         String var4;
-         while ((var4 = var3.readLine()) != null) {
-            var1.add(var4);
+         URLConnection connection = new URL(url.replaceAll(" ", "_")).openConnection();
+         connection.setRequestProperty("User-Agent", "Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.11 (KHTML, like Gecko) Chrome/23.0.1271.95 Safari/537.11");
+         connection.connect();
+         BufferedReader reader = new BufferedReader(new InputStreamReader(connection.getInputStream(), StandardCharsets.UTF_8));
+
+         String line;
+         while ((line = reader.readLine()) != null) {
+            lines.add(line);
          }
 
-         var3.close();
-         return var1;
-      } catch (Exception var5) {
+         reader.close();
+         return lines;
+      } catch (Exception ex) {
          return Collections.singletonList("4");
       }
    }
 
-   public static void open(Player var0, int var1) {
+   public static void open(Player player, int page) {
       if (!Values.m_aeMarket) {
-         var0.sendMessage("§cFeature is disabled.");
+         player.sendMessage("§cFeature is disabled.");
       } else {
          try {
-            Inventory var2 = Bukkit.createInventory(new MarketInventory.MarketInvHolder(), 54, "§a§lEnchantment Market");
-            if (var1 < 1) {
-               var0.sendMessage("§cPage §l" + var1 + "§c does not exist!");
+            Inventory inventory = Bukkit.createInventory(new MarketInventory.MarketInvHolder(), 54, "§a§lEnchantment Market");
+            if (page < 1) {
+               player.sendMessage("§cPage §l" + page + "§c does not exist!");
                return;
             }
 
-            byte var3 = 36;
-            int var4 = (var1 - 1) * var3;
-            int var5 = var1 * var3;
-            if (enchantHashmap.size() < var4) {
-               var0.sendMessage("§cPage §l" + var1 + "§c does not exist!");
+            byte itemsPerPage = 36;
+            int startIndex = (page - 1) * itemsPerPage;
+            int endIndex = page * itemsPerPage;
+            if (enchantHashmap.size() < startIndex) {
+               player.sendMessage("§cPage §l" + page + "§c does not exist!");
                return;
             }
 
-            if (enchantHashmap.size() < var5 && enchantHashmap.size() > var4) {
-               var5 = enchantHashmap.size();
+            if (enchantHashmap.size() < endIndex && enchantHashmap.size() > startIndex) {
+               endIndex = enchantHashmap.size();
             }
 
-            for (String var7 : enchantHashmap.isEmpty() ? new ArrayList() : new ArrayList<>(enchantHashmap.keySet()).subList(var4, var5)) {
-               EnchantPreview var8 = enchantHashmap.get(var7);
-               ItemBuilder var9 = new ItemBuilder(getItemStackDependingOnEnchantType(var8.getEnchantmentType()))
-                  .setAmount(var8.getLevelCount())
-                  .setName("§7Enchantment §6" + StringUtils.capitalize(var7.toLowerCase(Locale.ROOT)));
-               String[] var10 = var8.getDesc().contains("\n") ? var8.getDesc().split("\n") : var8.getDesc().split("7nl7");
-
-               for (String var14 : var10) {
-                  if (var14.chars().count() > 48L) {
-                     String[] var15 = var14.split(" ");
-                     int var16 = var15.length / 2;
-                     var9.addLoreLine("§e§lx §7" + collectArrayTogether(var15, 0, var16));
-                     var9.addLoreLine("§e§lx §7" + collectArrayTogether(var15, var16, var15.length - 1));
+            for (String enchantName : enchantHashmap.isEmpty() ? new ArrayList() : new ArrayList<>(enchantHashmap.keySet()).subList(startIndex, endIndex)) {
+               EnchantPreview preview = enchantHashmap.get(enchantName);
+               ItemBuilder itemBuilder = new ItemBuilder(getItemStackDependingOnEnchantType(preview.getEnchantmentType()))
+                  .setAmount(preview.getLevelCount())
+                  .setName("§7Enchantment §6" + StringUtils.capitalize(enchantName.toLowerCase(Locale.ROOT)));
+               String[] descLines = preview.getDesc().contains("\n") ? preview.getDesc().split("\n") : preview.getDesc().split("7nl7");
+
+               for (String descLine : descLines) {
+                  if (descLine.chars().count() > 48L) {
+                     String[] words = descLine.split(" ");
+                     int midpoint = words.length / 2;
+                     itemBuilder.addLoreLine("§e§lx §7" + collectArrayTogether(words, 0, midpoint));
+                     itemBuilder.addLoreLine("§e§lx §7" + collectArrayTogether(words, midpoint, words.length - 1));
                   } else {
-                     var9.addLoreLine("§e§lx §7" + var14);
+                     itemBuilder.addLoreLine("§e§lx §7" + descLine);
                   }
                }
 
-               var9.addLoreLine("");
-               var9.addLoreLine("§6§lx §7Popularity: §e" + var8.getDownloadCount() + " §eDownloads");
-               var9.addLoreLine("");
-               var9.addLoreLine("§6§lx §7Enchant Type: §e" + var8.getEnchantmentType());
-               var9.addLoreLine("§6§lx §7Applies to: §e" + var8.getAppliesTo());
-               var9.addLoreLine("§6§lx §7Levels: §e" + var8.getLevelCount());
-               var9.addLoreLine(" ");
-               var9.addLoreLine("§8>> §6Right-Click §7to §eVisit Author§7 on §6SpigotMc");
-               var9.addLoreLine("§8>> §6Left-Click §7to §eDownload");
-               ItemStack var20 = var9.toItemStack();
-               var20 = NBTapi.addNBTTag("enchant", var7, var20);
-               var20 = NBTapi.addNBTTag("page", var1 + "", var20);
-               var2.addItem(new ItemStack[]{var20});
+               itemBuilder.addLoreLine("");
+               itemBuilder.addLoreLine("§6§lx §7Popularity: §e" + preview.getDownloadCount() + " §eDownloads");
+               itemBuilder.addLoreLine("");
+               itemBuilder.addLoreLine("§6§lx §7Enchant Type: §e" + preview.getEnchantmentType());
+               itemBuilder.addLoreLine("§6§lx §7Applies to: §e" + preview.getAppliesTo());
+               itemBuilder.addLoreLine("§6§lx §7Levels: §e" + preview.getLevelCount());
+               itemBuilder.addLoreLine(" ");
+               itemBuilder.addLoreLine("§8>> §6Right-Click §7to §eVisit Author§7 on §6SpigotMc");
+               itemBuilder.addLoreLine("§8>> §6Left-Click §7to §eDownload");
+               ItemStack itemStack = itemBuilder.toItemStack();
+               itemStack = NBTapi.addNBTTag("enchant", enchantName, itemStack);
+               itemStack = NBTapi.addNBTTag("page", page + "", itemStack);
+               inventory.addItem(new ItemStack[]{itemStack});
             }
 
-            var2.setItem(
+            inventory.setItem(
                45,
                new ItemBuilder(Material.BOOK)
                   .setName("§6What is this?")
@@ -423,14 +424,14 @@
                   .toItemStack()
             );
 
-            for (int var19 : Arrays.asList(46, 47, 51, 52)) {
-               var2.setItem(var19, new ItemBuilder(AManager.matchMaterial("THIN_GLASS", 1, 0)).setName(" ").toItemStack());
+            for (int i : Arrays.asList(46, 47, 51, 52)) {
+               inventory.setItem(i, new ItemBuilder(AManager.matchMaterial("THIN_GLASS", 1, 0)).setName(" ").toItemStack());
             }
 
-            var2.setItem(48, new ItemBuilder(AManager.matchMaterial("BOOK", 1, 0)).setName("§8§ §6Previous Page").toItemStack());
-            var2.setItem(49, new ItemBuilder(Material.ANVIL, var1).setName("§6Close the selector.").toItemStack());
-            var2.setItem(50, new ItemBuilder(AManager.matchMaterial("BOOK", 1, 0)).setName("§8>> §6Next Page").setSkullOwner("MHF_ArrowRight").toItemStack());
-            var2.setItem(
+            inventory.setItem(48, new ItemBuilder(AManager.matchMaterial("BOOK", 1, 0)).setName("§8§ §6Previous Page").toItemStack());
+            inventory.setItem(49, new ItemBuilder(Material.ANVIL, page).setName("§6Close the selector.").toItemStack());
+            inventory.setItem(50, new ItemBuilder(AManager.matchMaterial("BOOK", 1, 0)).setName("§8>> §6Next Page").setSkullOwner("MHF_ArrowRight").toItemStack());
+            inventory.setItem(
                53,
                new ItemBuilder(AManager.matchMaterial("STORAGE_MINECART", 1, 0))
                   .setName("§6Share an enchantment")
@@ -449,58 +450,58 @@
                   )
                   .toItemStack()
             );
-            var0.openInventory(var2);
-         } catch (Exception var17) {
-            var17.printStackTrace();
-            var0.sendMessage("§c§lERROR:§c Please wait until the market is cached in your system.");
+            player.openInventory(inventory);
+         } catch (Exception ex) {
+            ex.printStackTrace();
+            player.sendMessage("§c§lERROR:§c Please wait until the market is cached in your system.");
          }
       }
    }
 
-   private static String collectArrayTogether(String[] var0, int var1, int var2) {
-      StringBuilder var3 = new StringBuilder();
+   private static String collectArrayTogether(String[] arr, int startFrom, int endWith) {
+      StringBuilder stringBuilder = new StringBuilder();
 
-      for (int var4 = var1; var4 < var2; var4++) {
-         var3.append(var0[var4] + " ");
+      for (int i = startFrom; i < endWith; i++) {
+         stringBuilder.append(arr[i] + " ");
       }
 
-      return var3.toString();
+      return stringBuilder.toString();
    }
 
-   private static ItemStack getItemStackDependingOnEnchantType(String var0) {
+   private static ItemStack getItemStackDependingOnEnchantType(String at) {
       return new ItemStack(Material.BOOK);
    }
 
-   public static String readTextFile(String var0) {
-      StringBuilder var1 = new StringBuilder();
+   public static String readTextFile(String fileName) {
+      StringBuilder returnValue = new StringBuilder();
 
       try {
-         FileReader var2 = new FileReader(new File(Core.getInstance().getDataFolder(), var0));
+         FileReader file = new FileReader(new File(Core.getInstance().getDataFolder(), fileName));
 
-         String var3;
-         try (BufferedReader var4 = new BufferedReader(var2)) {
-            while ((var3 = var4.readLine()) != null) {
-               var1.append(var3).append("\n");
+         String line;
+         try (BufferedReader reader = new BufferedReader(file)) {
+            while ((line = reader.readLine()) != null) {
+               returnValue.append(line).append("\n");
             }
          }
-      } catch (FileNotFoundException var9) {
+      } catch (FileNotFoundException ex) {
          throw new RuntimeException("File not found");
-      } catch (IOException var10) {
+      } catch (IOException ex) {
          throw new RuntimeException("IO Error occurred");
       }
 
-      return var1.toString();
+      return returnValue.toString();
    }
 
-   public static void writeTextFile(String var0, String var1) {
+   public static void writeTextFile(String fileName, String s) {
       try {
-         FileWriter var2 = new FileWriter(new File(Core.getInstance().getDataFolder(), var0));
-         BufferedWriter var3 = new BufferedWriter(var2);
-         var3.write(var1);
-         var3.close();
-         var2.close();
-      } catch (IOException var4) {
-         var4.printStackTrace();
+         FileWriter output = new FileWriter(new File(Core.getInstance().getDataFolder(), fileName));
+         BufferedWriter writer = new BufferedWriter(output);
+         writer.write(s);
+         writer.close();
+         output.close();
+      } catch (IOException ex) {
+         ex.printStackTrace();
       }
    }
 
